### PR TITLE
Move Badge component to @automattic/components

### PR DESCRIPTION
**Pull request by valterlorran**
Related to #77564
* https://github.com/Automattic/wp-calypso/pull/78809

## Proposed Changes

* Copies the `Badge` component and related tests, examples, etc. to `@automattic/components` so we can use it in other packages (like Launchpad).
* References to Badge to the new package will be updated on https://github.com/Automattic/wp-calypso/pull/78812.
* The original PR(linked above) was reverted due to an error executing `yarn build-packages`. It would fail to find the `PropTypes` while building the packages:
https://github.com/Automattic/wp-calypso/blob/1a10a9a4892d77df834f8c0667ef6adc1df6e909/packages/components/src/badge/index.jsx#L9
  * More context: p1688481304807469-slack-C02DQP0FP  

## Testing Instructions

* Make sure all tests pass.
```bash
yarn run test-packages packages/components/src/badge/test/index.js
```
* Run `yarn build-packages` and make sure it works
* This alone should not break anything since it's making a copy of the files in question. :) 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by daledupreez** [comment]
@valterlorran, I am concerned that this may lead to duplicate code being sent down to users (especially CSS), and that may lead to subtle bugs. I don't think it's likely, as any code overriding the CSS should be using more specific selectors, but it may occur.

Might it be worth making the existing `client/components/badge` a shim/wrapper around the new code? We could also tackle that in a follow-up, but I feel that we should minimise the amount of time where we have duplicated code.

**Comment by valterlorran** [comment]
@daledupreez Sorry, but I don't understand how that can be possible if it's not being used anywhere yet 🤔 I intended to work on https://github.com/Automattic/wp-calypso/pull/78812, which will remove the `client/components/badge` and replace the usages with the package included in this PR.

**Comment by daledupreez** [comment]
The missing piece for me was #78812, which captures the work to switch from `client/components/badge` to `@automattic/components` - could you add that in the description for this PR to call out that the follow-up work will be tackled in that PR?

I would still prefer that we update the existing code in `client/components/badge` to be a thin wrapper around the new component from this PR, as that allows us to handle the migrations in a systemic way that's easier to test. But that's a discussion for the follow-up work rather than this PR. 😁

File: packages/components/src/badge/index.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import classNames from 'classnames';
+import { ReactNode } from 'react';
```
Comment (daledupreez) [inline packages/components/src/badge/index.tsx]: Nit: this should be a type import.
```suggestion
import type { ReactNode } from 'react';
```

File: packages/components/src/badge/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+function MyComponent() {
+	return <Badge type="warning">Only 6MB left!</Badge>;
+}
+```
+
+## Props
+
+The following props are available to customize the Badge:
+
+- `type`: `'warning'`, `'success'`, `'info'`, `'info-blue'`, `'error'`
```
Comment (daledupreez) [inline packages/components/src/badge/README.md:20]: Should we document the `className` prop as well?

**Review by daledupreez — APPROVED** [review]
I do have some minor comments, but this LGTM overall.

**Pull request merged**